src/resources/FreeKill.ts

```diff
@@ -0,0 +1,167 @@
+import { mallPrice, myLightning, restoreMp, retrieveItem, use } from "kolmafia";
+
+import { Macro } from "../combat";
+import { have } from "../lib";
+import { Requirement } from "../maximize";
+import { get } from "../property";
+import { $familiar, $item, $items, $skill } from "../template-string";
+import {
+  ActionSource,
+  findActionSource,
+  FindActionSourceOptions,
+} from "./action";
+import * as AsdonMartin from "./2017/AsdonMartin";
+
+const freeKillSources: ActionSource[] = [
+  // Free limited sources
+  new ActionSource(
+    $skill`Gingerbread Mob Hit`,
+    () =>
+      !get("_gingerbreadMobHitUsed") && have($skill`Gingerbread Mob Hit`)
+        ? 1
+        : 0,
+    Macro.skill($skill`Gingerbread Mob Hit`),
+    {
+      preparation: () => restoreMp(30),
+    }
+  ),
+
+  new ActionSource(
+    $skill`Shattering Punch`,
+    () =>
+      have($skill`Shattering Punch`) ? 3 - get("_shatteringPunchUsed") : 0,
+    Macro.skill($skill`Shattering Punch`),
+    {
+      preparation: () => restoreMp(30),
+    }
+  ),
+
+  new ActionSource(
+    $item`replica bat-oomerang`,
+    () =>
+      have($item`replica bat-oomerang`)
+        ? 3 - get("_usedReplicaBatoomerang")
+        : 0,
+    Macro.item($item`replica bat-oomerang`)
+  ),
+
+  new ActionSource(
+    $item`The Jokester's gun`,
+    () =>
+      !get("_firedJokestersGun") && have($item`The Jokester's gun`) ? 1 : 0,
+    Macro.skill($skill`Fire the Jokester's Gun`),
+    {
+      equipmentRequirements: () =>
+        new Requirement([], {
+          forceEquip: $items`The Jokester's gun`,
+        }),
+    }
+  ),
+
+  new ActionSource(
+    $item`Lil' Doctor™ bag`,
+    () => (have($item`Lil' Doctor™ bag`) ? 3 - get("_chestXRayUsed") : 0),
+    Macro.skill($skill`Chest X-Ray`),
+    {
+      equipmentRequirements: () =>
+        new Requirement([], {
+          forceEquip: $items`Lil' Doctor™ bag`,
+        }),
+    }
+  ),
+
+  new ActionSource(
+    $skill`Asdon Martin: Missile Launcher`,
+    () => (!get("_missileLauncherUsed") && AsdonMartin.have() ? 1 : 0),
```

Comment:
I think we want .installed(), not .have(); if not, we should account for whether your workshed was changed today. This likely applies to the bumperbanish too,